pkg/fleets/controller.go

```diff
@@ -233,6 +241,18 @@ func (c *Controller) loggerForFleet(f *agonesv1.Fleet) *logrus.Entry {
 	return c.loggerForFleetKey(fleetName).WithField("fleet", f)
 }
 
+func loggerForFleetKey(key string) *logrus.Entry {
```

Comment:
These functions are duplicates of ones that already exist. Instead of copying them, how about extracting the existing functions to live outside of the controller struct and taking the logger as an argument (since that is the only field in the controller that they need)? You may need to change a lot of call sites, but it will prevent code duplication here and also make the function have a clear and small surface and purpose.

Reply:
Removed the duplicates and now takes in the logger as a param. Thanks!